### DISCUSSION TITLE
How to write in parallel

### DISCUSSION THREAD

**Discussion by zhyxue0** [firedrakeproject/firedrake#2679] [Firedrake support] [answered]
Hi, everyone. I'm trying to solve poisson's equation in parallel(`mpiexec -n 4 python poisson_mixed.py`), the code `File("poisson_mixed.pvd").write(u)` seems to be the same as the serial version. I want to write solution ` u` to txt file under parallel version, but the code ` np.savetxt(file_name, u.dat.data)` will write repeatedly on different processors. How to implement it in parallel?

**Comment by wence-** [comment, marked as answer]
[I moved this to a discussion since I think that's probably a better place for it]

What would you like to do with the text file after you write it? Just writing the coefficients loses a lot of information, especially in parallel where the distribution of the mesh between processes is not known.

If you need this for checkpointing, consider using [`CheckpointFile`](https://firedrakeproject.org/checkpointing.html#checkpointing-with-checkpointfile) which is designed for this purpose.

**Reply by zhyxue0** [reply]
Thanks. I just want to prevent the program from crashing during running.
